src/components/MultiHopTrade/hooks/useQuoteValidationErrors.ts

```diff
@@ -0,0 +1,78 @@
+import { useHopHelper } from 'components/MultiHopTrade/hooks/useHopHelper'
+import { useIsTradingActive } from 'components/MultiHopTrade/hooks/useIsTradingActive'
+import { useReceiveAddress } from 'components/MultiHopTrade/hooks/useReceiveAddress'
+import { ActiveQuoteStatus } from 'components/MultiHopTrade/types'
+import { useWallet } from 'hooks/useWallet/useWallet'
+import { walletSupportsChain } from 'hooks/useWalletSupportsChain/useWalletSupportsChain'
+import { bnOrZero } from 'lib/bignumber/bignumber'
+import { isTruthy } from 'lib/utils'
+import {
+  selectFirstHopNetworkFeeCryptoPrecision,
+  selectFirstHopSellAsset,
+  selectFirstHopTradeDeductionCryptoPrecision,
+  selectLastHopBuyAsset,
+  selectLastHopNetworkFeeCryptoPrecision,
+  selectMinimumSellAmountCryptoBaseUnit,
+  selectSellAmountCryptoBaseUnit,
+} from 'state/slices/tradeQuoteSlice/selectors'
+import { useAppSelector } from 'state/store'
+
+type QuoteValidationErrors = ActiveQuoteStatus[]
+
+export const useQuoteValidationErrors = (): QuoteValidationErrors => {
+  const {
+    sellAssetBalanceCryptoBaseUnit,
+    firstHopFeeAssetBalancePrecision,
+    lastHopFeeAssetBalancePrecision,
+  } = useHopHelper()
+  const wallet = useWallet().state.wallet
+  const { isTradingActiveOnSellPool, isTradingActiveOnBuyPool } = useIsTradingActive()
+
+  const sellAmountCryptoBaseUnit = useAppSelector(selectSellAmountCryptoBaseUnit)
+  const firstHopNetworkFeeCryptoPrecision = useAppSelector(selectFirstHopNetworkFeeCryptoPrecision)
+  const lastHopNetworkFeeCryptoPrecision = useAppSelector(selectLastHopNetworkFeeCryptoPrecision)
+  const firstHopTradeDeductionCryptoPrecision = useAppSelector(
+    selectFirstHopTradeDeductionCryptoPrecision,
+  )
+  const minimumSellAmountBaseUnit = useAppSelector(selectMinimumSellAmountCryptoBaseUnit)
+  const firstHopSellAsset = useAppSelector(selectFirstHopSellAsset)
+  const lastHopBuyAsset = useAppSelector(selectLastHopBuyAsset)
+  const receiveAddress = useReceiveAddress()
+
+  const walletSupportsSellAssetChain = walletSupportsChain({
+    chainId: firstHopSellAsset?.chainId ?? '',
+    wallet,
+  })
+  const walletSupportsBuyAssetChain = walletSupportsChain({
+    chainId: lastHopBuyAsset?.chainId ?? '',
+    wallet,
+  })
```

Comment:
my brain cant cope with passing in empty string 😅
```suggestion
  const walletSupportsSellAssetChain = firstHopSellAsset && walletSupportsChain({
    chainId: firstHopSellAsset.chainId,
    wallet,
  })
  const walletSupportsBuyAssetChain = lastHopBuyAsset && walletSupportsChain({
    chainId: lastHopBuyAsset.chainId,
    wallet,
  })
```

Reply:
That's fair, it's a bad habit this codebase has, which I also hate, that I shouldn't be propagating.

https://github.com/shapeshift/web/pull/4889/commits/3c2dfb0db4a92e0f1a79c3401a5342d15ab69a42